### ISSUE TITLE
Question on vendoring dependencies for CF with --no-binary flag

### ISSUE THREAD

**Issue by mguarascio** [cloudfoundry/python-buildpack#108]
What version of Cloud Foundry and CF CLI are you using? (i.e. What is the output of running cf curl /v2/info && cf version?
{
"name": "",
"build": "",
"support": "https://support.pivotal.io",
"version": 0,
"description": "",
"authorization_endpoint": "",
"token_endpoint": "",
"min_cli_version": "6.23.0",
"min_recommended_cli_version": "6.23.0",
"api_version": "2.82.0",
"app_ssh_endpoint": "",
"app_ssh_host_key_fingerprint": "",
"app_ssh_oauth_client": "ssh-proxy",
"routing_endpoint": "",
"doppler_logging_endpoint": ""
}

What version of the buildpack you are using?
1.5.26

If you were attempting to accomplish a task, what was it you were attempting to do?
Package dependencies, following the guidelines here by including the --no-binary :all: flags: https://docs.cloudfoundry.org/buildpacks/python/index.html#vendoring

What is the recommendation for vendoring dependencies when a package ONLY includes binaries in their latest releases? For example, mongomock 3.10 only has a wheel file. When I do pip install mongomock I get 3.10, but by including the --no-binary flags I get a much older release:

```
pip install mongomock --no-binary :all:
Successfully installed mongomock-3.0.0 sentinels-1.0.0 six-1.11.0
```

When I drop the --no-binary :all:, I get the latest since it has only the wheel file (binary):

```
pip uninstall mongomock
pip install mongomock
Successfully installed mock-2.0.0 mongomock-3.10.0 pbr-4.0.3 
```

Please confirm where necessary:
* [ ] I have included a log output
* [ ] My log includes an error message
* [ ] I have included steps for reproduction

**Comment by sclevine** [comment]
Apologies for the delay.

We recommend `--no-binary :all:` so that you don't push the app with native extensions that are pre-compiled for the wrong platform.

Can you see if vendoring the app with Linux binaries solves this problem? 
```
pip install mongomock --platform linux_x86_64
```

Have you experienced this with other packages besides mongomock?

(Also, I recommend upgrading to a recent version of the buildpack. The buildpack has been entirely re-written since 1.5.26.)

**Comment by mguarascio** [comment]
Ok thank you for the information. So far I've only seen it for this package, based on feedback from our internal customers. And we were able to get by with the precompiled version, which appears to be compatible with all platforms. 

In regards to the version - yes we recently made the latest version available in our environments (1.6.15), although we do have some customers that need to upgrade to that one.
